### DISCUSSION TITLE
Command Palette

### DISCUSSION THREAD

**Discussion by matthewbirtch** [mattermost/community#31] [Enhancements]
The idea of a command palette is a common pattern among dev IDEs and I think there is potential that the Mattermost platform could be enhanced with such a feature. The Command Palette could support things like:
- Search (messages, files, channels, people, etc)
- Slash commands - having a global place to run slash commands could be great. Slash commands don't always make sense in the channel context (e.g. adding todos) 
- Open specific settings
- Open other products from the suite
- Installing Apps/Plugins
 
![image](https://user-images.githubusercontent.com/2040554/122956720-3fba8e80-d34f-11eb-8738-c0d82313336b.png)

**Comment by andrewbrown00** [comment]
Love that you added this @matthewbirtch, I'd really like to see this feature get traction! Maybe a candidate for an open-source Friday collaboration?

**Reply by larkox** [reply]
If I remember correctly, @levb has been thinking about this for a while. Sounds like a nice idea.
